scss/_patterns_side-navigation.scss

```diff
@@ -0,0 +1,104 @@
+@import 'settings';
+
+@mixin vf-p-side-navigation {
+  .p-side-navigation__list {
+    @extend %vf-list;
+
+    .p-side-navigation > &:first-child {
+      // nudge from top to put text on baseline grid
+      padding-top: 0.5 * $spv-inner--x-small;
+    }
+
+    & + & {
+      @extend %vf-pseudo-border--top;
+
+      &::after {
+        left: $sph-inner;
+        top: -0.5 * $spv-outer--scaleable; // place border in the middle of the margin
+      }
+    }
+  }
+
+  .p-side-navigation__item--title {
+    font-weight: $font-weight-bold;
+  }
+
+  .p-side-navigation__text,
+  .p-side-navigation__link {
+    display: block;
+    padding: $spv-inner--x-small $sph-inner--small $spv-inner--x-small $sph-inner;
+
+    &.is-active,
+    &.is-selected {
+      font-weight: $font-weight-bold;
+    }
+
+    // nested 2 or 3 levels
+    .p-side-navigation__item .p-side-navigation__item & {
+      padding-left: 2 * $sph-inner;
+    }
+
+    .p-side-navigation__item .p-side-navigation__item .p-side-navigation__item & {
+      padding-left: 3 * $sph-inner;
+    }
+  }
+
+  .p-side-navigation__link {
+    @include vf-focus;
+
+    &:hover {
+      text-decoration: none;
+    }
+  }
+
+  // default light theme
+  @include vf-side-navigation-theme;
+}
+
+// theme
+@mixin vf-side-navigation-theme(
+  // default text color in sidebar
+    $color-sidenav-text-default: $color-mid-dark,
```

Comment:
When it comes to colours, in my opinion these local vars should point to the new [theme colour vars](e.g. `$colors--light-theme--border-default`). They are named after what they're supposed to be used for, and allow us to audit colour use easier. With the generic ones, it's difficult to know what they were meant for, or what is the other theme's counterpart for a specific colour.

Reply:
so this line would be: 
`$color-sidenav-text-default: $colors--light-theme--text-default;`

Reply:
Sure, thanks for pointers. Will update that.

Reply:
@lyubomir-popov So, for side nav we actually use lighter text, so default text will use `--text-disabled` and highlighted version will use `--text-default`.

Reply:
Is this related to https://github.com/canonical-web-and-design/vanilla-framework/issues/2851?

Reply:
@spencerbygraves A little bit I guess. I think the only place where we used 'faded out' text colour was disabled text, so the name of the variable reflects that.